contributors/design-proposals/api-proposal-design-template.md

```diff
@@ -0,0 +1,143 @@
+
+## Requirements
+
+Bullet point list of new API requirements.  Requirements will in part be driven by use
+cases, but may have additional engineering or interoperability requirements:
+
+- User requirement
+- Engineering requirement
+- Interoperability requirement
+
+## Overview subsequent versions
+
+Subsequent versions will require their own API proposals, however it is important to include
+the plans for them here to provide a broader picture.
+
+### Beta
+
+*Use cases*
+
+- Use cases for transitioning to Beta
+
+*Requirements*
+
+- Additional requirements for transitioning to Beta
+
+### GA
+
+*Use cases*
+
+- Use cases for transitioning to Beta
+
+*Requirements*
+
+- Additional requirements for transitioning to Beta
+
+## Dependencies and deadlines
+
+List of features that will depend on these changes.
+
+List of features that these changes will depend upon.
+
+**Include any deadlines that are related.** e.g. another feature depends on this and it is slated for 1.x.
+
+## Proposed changes for new types or existing types
+
+Declare the proposed Group / Version / Kind for any new types created as part of this proposal.
+
+Declare any new fields for existing types and their version.  Will the fields be added as first class fields or
+as annotations?
+
+Describe how any new types or fields will be used by the end user.  Include sample yaml config and full
+walkthrough examples of 1 or more usecases.
+
+### Defaulting
+
+Describe defaulting that will be done
+
+### Validation
+
+Describe validation the will be done
+
+### Patching lists - strategic merge keys
+
+For any list fields, describe whether they will be replaced when patched or merged.  If merged,
+describe the merge key that will be used.
+
+**mergeKeys must be unique and required for all elements in the list!**  Consequences of using a
+non-unique or optional merge key may be severe.
+
+### Controllers
+
+Describe any controllers that will be added or updated.
+
+### Subresources
+
+Describe any subresources that will be added.
+
+## Mandatory API pre-flight check list
+
+The following items must be considered and explained when adding or changing and API.
+
+### API convention deviations
+
+List and justify any deviations from [API conventions].  *None* is acceptable if
+there are no deviations.
+
+### Impact on backwards / forwards compatibility
+
+Describe any impact on backwards / forwards [compatibility].
+
+Known future incompatibility risks and mitigation.
+
+### Security, PII and authz
+
+Describe any security concerns or PII that is managed by the API.  Describe
+any special considerations that must be made for authentication or
+authorization.
+
+### Cli and client library considerations.
+
+Describe any interactions with the cli or client libraries.
+
+Is functionality in this API already present in the cli?  If so, was is the interaction and
+path forward?
+
+Is server side garbage collection needed / enabled?
+
+## Alternatives considered
```

Comment:
It seems to me that some of the items in this list really belong in a proposal or design doc. I would consider focusing this doc on things that are really unique to the api, as it is feeling a little heavyweight.